### PR TITLE
WT-7989 Compact quits when running at the same time as a system checkpoint

### PR DESCRIPTION
**Pull request by AlexeyAnisimovAU**
The scope of this change is to implement a test that reproduces the issue.

### CODE REVIEW

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:31]: Typo `to` => `two`

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:34]: Typo `test` => `tests`

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:42]: Missing space `stringto`

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:112]: You don't need the brackets here

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:208]: You don't need the brackets here

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:229]: You don't need the brackets here

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:60]: You may want to alphabetize that part, just a suggestion though.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:276]: Missing brackets around the returned value

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:129]: range. To let -> range to let

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:150]: add comment : Wait for the threads to finish work.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:164]: I am not sure how other csuite tests normally do this but I believe you can print out the file sizes before and after unconditionally and then assert.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:187]: I think the use of `td->session_cond` is incorrect. The API doesn't ask for the session pointer for the thread waiting for this condition variable. I believe we just need to pass the `session` from the current thread.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:141]: `td.session_cond` is not required.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:56]: No need to pass `*session_cond` variable.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:252]: Add a comment about the range of keys being deleted (i.e., middle 33%).

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:184]: Add a comment about how much this delay is.

**Review by hbokhari — COMMENTED** [review]
@AlexeyAnisimovAU - can you please add this to evergreen as well? We have to test this before doing a final approval.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:31]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:34]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:42]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:112]: There's a multiline comment in the block. The brackets are there to improve readability.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:208]: Removed the brackets.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:229]: Removed the brackets.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:60]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:276]: Why do we need them here?

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:129]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:150]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:164]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:252]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:184]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [reply test/csuite/wt7989_compact_checkpoint/main.c:276]: I believe this is part of our coding guidelines.

Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:276]: Done. Thank you!

File: test/csuite/CMakeLists.txt
Comment (etienneptl) [inline test/csuite/CMakeLists.txt:198]: This should be added after line 298 to keep the order correct.

File: test/csuite/Makefile.am
Comment (etienneptl) [inline test/csuite/Makefile.am:10]: This should be moved below too to keep the order

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:169]: Should we add a FIXME-ticketID here with the ticket that fixes the problem ?

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:192]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:140]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:55]: Done

File: test/csuite/CMakeLists.txt
Comment (AlexeyAnisimovAU) [reply test/csuite/CMakeLists.txt:198]: Oops, I forgot to move it after renamed the test. Good catch, thanks!

File: test/csuite/Makefile.am
Comment (AlexeyAnisimovAU) [reply test/csuite/Makefile.am:10]: Done, thanks!

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:169]: Done, thanks!

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:138]: You don't need the curly brackets here, same for the else just below.
Or you could keep them and merge what's below, from line 145.

Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Good point, done!

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:216]: Is it good practice to wait indefinitely?

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:188]: Missing a space at the end of the comment.

**Review by etienneptl — APPROVED** [review]
LGTM, with one question about the `__wt_cond_wait_signal` function.

**Review by raviprakashgiri29 — APPROVED** [review]
LGTM

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Done.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Added 20 seconds timeout.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:228]: Missing a `s` at the end of `second`

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:212]: Missing brackets

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:228]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:212]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:188]: Multi-line comment not formatted correctly.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:210]: Multi-line comment not formatted correctly.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (etienneptl) [inline test/csuite/wt7989_compact_checkpoint/main.c:228]: Multi-line comment not formatted correctly.

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Done

File: test/csuite/wt7989_compact_checkpoint/main.c
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c:188]: Done

**Pull request merged**
